bindings.c

```diff
@@ -3517,18 +3518,32 @@ static int proc_meminfo_read(char *buf, size_t size, off_t offset,
 		} else if (startswith(line, "MemAvailable:")) {
 			snprintf(lbuf, 100, "MemAvailable:   %8lu kB\n", memlimit - memusage + cached);
 			printme = lbuf;
-		} else if (startswith(line, "SwapTotal:") && memswlimit > 0) {
-			sscanf(line+sizeof("SwapTotal:")-1, "%lu", &hostswtotal);
-			if (hostswtotal < memswlimit)
-				memswlimit = hostswtotal;
-			snprintf(lbuf, 100, "SwapTotal:      %8lu kB\n", memswlimit);
-			printme = lbuf;
-		} else if (startswith(line, "SwapFree:") && memswlimit > 0 && memswusage > 0) {
-			unsigned long swaptotal = memswlimit,
+		} else if (startswith(line, "SwapTotal:")) {
+                        if (memswlimit > 0) {
```

Comment:
Please collapse these cases nicer so that it's clear to the reader what's going on.  Since the second case will overwrite the first and memswlimit and hostswtotal are not used elsewhere this isn't wrong, but the reader has to go looking to make sure that those are in fact the case.

Actually, do you want the SwapTotal: line to show up in the swap_off == true case?

In either case, perhaps just

```
if (opts->swap_off == true)
   memswlimit = 0;
if (memswlimit > 0) { ... }

Reply:
I'm trying to rewrite the unsightly code right now. I wasn't happy with the implementation either.
To do that, I needed to understand the existing code even better. 
I came across a question. Is the order of the entries in "meminfo" fixed?
The reason for my question:
The variable "memswlimit" is set before the loop that processes the entries in "meminfo" starts.
In the line "SwapTotal" the value of "memswlimit" may be changed.
 if (hostswtotal < memswlimit)
              memswlimit = hostswtotal;
In the "SwapFree" line, the value of "memswlimit" is used in a formula.

Thus the result can change depending on the order of the entries "SwapTotal" and "SwapFree".

Reply:
If you look at fs/proc/meminfo.c in the kernel, you can see

```
        show_val_kb(m, "SwapTotal:      ", i.totalswap);
        show_val_kb(m, "SwapFree:       ", i.freeswap);
```
so the order is 'fixed', but there is no guarantee about the order (that is, a future code change can change it), so your concern is valid.